openadapt/record.py

```diff
@@ -1128,7 +1188,115 @@ def audio_callback(
 
 @logger.catch
 @utils.trace(logger)
-def record(
+async def read_browser_events(
+    websocket: websockets.WebSocketServerProtocol,
+    path: str,
+    event_q: queue.Queue,
+    terminate_processing: Event,
+    recording: Recording,
+    started_counter: multiprocessing.Value,
+) -> None:
+    """Read browser events and add them to the event queue.
+
+    Params:
+        websocket: The websocket object.
+        path: The path to the websocket.
+        event_q: A queue for adding browser events.
+        terminate_processing: An event to signal the termination of the process.
+        recording: The recording object.
+        started_counter: Value to increment once started.
+
+    Returns:
+        None
+    """
+    utils.set_start_time(recording.timestamp)
+
+    logger.info("Starting Reading Browser Events ...")
+    started = False
+
+    while not terminate_processing.is_set():
+        async for message in websocket:
+            browser_data = message
+
+            if not browser_data:
+                continue
+
+            if not started:
+                with started_counter.get_lock():
+                    started_counter.value += 1
+                started = True
+
+            timestamp = utils.get_timestamp()
+            event_tuple = Event(
+                timestamp,
+                "browser",
+                {"message": browser_data},
+            )
+            event_q.put(event_tuple)
+
+
+@logger.catch
+@utils.trace(logger)
+async def start_browser_event_server(
+    event_q: queue.Queue,
+    terminate_processing: Event,
+    recording: Recording,
+    started_counter: multiprocessing.Value,
+) -> None:
+    """Start the browser event server.
+
+    Params:
+        event_q: A queue for adding browser events.
+        terminate_processing: An event to signal the termination of the process.
+        recording: The recording object.
+        started_counter: Value to increment once started.
+
+    Returns:
+        None
+    """
+    server = await websockets.serve(
+        lambda ws, path: read_browser_events(
+            ws, path, event_q, terminate_processing, recording, started_counter
+        ),
+        "localhost",
+        8765,
```

Comment:
What do you think about moving this to `config`?

Reply:
Done here: 3836c3d165bb11c5615c83a0df65f6cdc155f93f